### PR TITLE
tests: repair tests

### PR DESCRIPTION
**Pull request by sabinem**
This PR does the following things:

- unsupported px keywords are no longer reported as to not confuse the user
- the `encoding` argument is also needed when checking the px file
- the tests were repaired, so they run through again

### CODE REVIEW

File: R/scan_px_file.R [hunk 1]
```diff
@@ -37,17 +37,20 @@ scan_px_file <- function(
...
-        file_or_url, what = "list", sep = ";", quote = NULL,
-        quiet = TRUE, encoding = encoding, multi.line = TRUE)
+        file_or_url,
+        what = "list",
+        sep = ";",
+        quote = NULL,
+        quiet = TRUE,
+        encoding = encoding,
+        multi.line = TRUE)
     },
```
Comment (sabinem) [inline R/scan_px_file.R:54]: Here just the formatting was changed

File: R/scan_px_file.R [hunk 2]
```diff
@@ -37,17 +37,20 @@ scan_px_file <- function(
   tryCatch(
     {
       # check px file format and get back default language and available locales
-      file <- check_file_or_url(file_or_url, locale)
-
+      file <- check_file_or_url(file_or_url, locale, encoding)
```
Comment (sabinem) [inline R/scan_px_file.R:40]: The file check also needs the encoding to read the file.

File: R/parse_px_lines.R [hunk 3]
```diff
@@ -49,8 +49,6 @@ parse_px_lines <- function(
           # make sure the line is not empty
           if (is_supported_px_key(px_key, supported_keywords)) {
             px_rows <- append(px_rows, get_keyword_values_pair(px_line_group))
-          } else {
-            logger::log_info(paste("unsupported keyword detected", px_key))
```
Comment (sabinem) [inline R/parse_px_lines.R:53]: The unsupported keywords are no longer reported.

File: R/check_file_or_url.R [hunk 4]
```diff
@@ -17,18 +18,16 @@
...
+                         encoding = encoding)
       keywords <- get_keywords_from_lines(lines)
-      if (!("AXIS-VERSION" %in% keywords)) {
-        stop("File is not a px cube: could not find AXIS-VERSION statement")
-      }
-      if (!('CHARSET=\"ANSI\";' %in% lines)) {
-        stop("File has not the expected ANSI encoding: 
-              file an issue and ask for an extensions to other encodings")
+      if (!("MATRIX" %in% keywords)) {
+        stop("File is not a px cube: could not find mandatory MATRIX statement")
```
Comment (sabinem) [inline R/check_file_or_url.R:30]: MATRIX is a mandatory px keyword, while the previously used keyword are not mandatory.

File: R/check_file_or_url.R [hunk 5]
```diff
@@ -38,9 +37,10 @@ check_file_or_url <- function(
           stop("File has translations but no default language.")
         }
         language_pattern <- get_language_pattern_for_keys(languages)
-        file <- list(is_multilingual = TRUE, languages = languages,
-                             language_pattern = language_pattern,
-                             default_language = default_language)
+        file <- list(is_multilingual = TRUE,
+                     languages = languages,
+                     language_pattern = language_pattern,
+                     default_language = default_language)
```
Comment (sabinem) [inline R/check_file_or_url.R:43]: only reformatting happened here.

**Review by cmdoret — APPROVED** [review]
These changes look good to me!

**Comment by cmdoret** [comment]
One general comment about parse_px_lines (not specific to this PR): The conditions are nested pretty deep, which makes it a bit hard to figure out the flow: 
https://github.com/SDSC-ORD/pxRRead/blob/842f4e835bf23967b9b9e02df519fedc7202bfbd/R/parse_px_lines.R#L34-L55

Would it make sense to use `next` here to reduce nesting? Something like:

```r
if (scanned_lines[i] != "") { 
   px_line_group <- append(px_line_group, scanned_lines[i])
   next
}
if (length(px_line_group) < 0) {
   next
} 
# the key is contained in the first row of the line group
px_key <- get_keywords_from_lines(px_line_group[1])
...
```
I feel this could help with readability, but I am not sure, just a suggestion.

**Comment by sabinem** [comment]
I have followed you suggestion and thanks to the tests it was easy to be sure that no bug was introduced by this change.

**Pull request merged**